runelite-client/src/main/java/net/runelite/client/plugins/nightmarezone/NightmareZonePlugin.java

```diff
@@ -112,14 +121,47 @@ public void onGameTick(GameTick event)
 			if (!absorptionNotificationSend)
 			{
 				absorptionNotificationSend = true;
+				lastTickMillis = 0;
+				gameTime = 0;
+				pointsPerHour = 0;
 			}
 
+			if (!hitpointsNotificationSend)
+			{
+				hitpointsNotificationSend = true;
+			}
 			return;
 		}
 		if (config.absorptionNotification())
 		{
 			checkAbsorption();
 		}
+
+		if (config.hitpointsNotification())
+		{
+			checkHitpoints();
+		}
+	}
+
+	@Schedule(
+		period = 1,
+		unit = ChronoUnit.SECONDS
+	)
+	public void tickSkillTimes()
+	{
+		if (isInNightmareZone())
+		{
+			if (lastTickMillis == 0)
+			{
+				lastTickMillis = System.currentTimeMillis();
+				return;
+			}
+			final long nowMillis = System.currentTimeMillis();
+			final long tickDelta = nowMillis - lastTickMillis;
+			lastTickMillis = nowMillis;
+			gameTime += tickDelta;
+			pointsPerHour = 3600 * client.getVar(Varbits.NMZ_POINTS) / (gameTime / 1000);
```

Comment:
Something like this can just be calculated in the overlay by storing start tick and start points.